deploy/examples/create-external-cluster-resources.py

```diff
@@ -652,6 +644,20 @@ def __init__(self, arg_list=None):
             self.cluster = rados.Rados()
             self.cluster.conf_read_file()
         self.cluster.connect()
+        if not self._arg_parser.rgw_pool_prefix and not self._arg_parser.upgrade:
+            self._arg_parser.rgw_pool_prefix = self.DEFAULT_RGW_POOL_PREFIX
+        # when user has not provided rgw pool name during upgrade, get the rgw pool name used during connection
+        if not self._arg_parser.rgw_pool_prefix and self._arg_parser.upgrade:
+            _, val = self.check_user_exist("client.healthchecker")
```

Comment:
can we move this check to `    def upgrade_user_permissions(self, user): ` where we already get the user caps

Reply:
The reason I kept under `_init` method is this is the first method called by python and here we are updating the cli flag `self._arg_parser.rgw_pool_prefix` so kept in this method. Also, other flags are modified here only.

Reply:
Actually this is just for an upgrade and I feel like we should make this separate and maybe even a separate function for this